Mathlib/Tactic/Widget/Explain.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import Mathlib.Init
+import ProofWidgets.Component.HtmlDisplay
+
+/-!
+
+# Explanation Widgets
+
+This file defines some simple widgets wrapped in a tactic, term and command elaborator
+that display nicely rendered markdown in the infoview.
+
+Example tactic usage:
+```
+example : True := by
+  explain "This is the first step."
+  explain "This is the last step." in
+    exact trivial
+```
+Placing the cursor on each line results will render the explanation in the infoview.
+
+Example term usage:
+```
+example : Nat := explain "This is zero" in
+  0
+```
+Placing the cursor on the term will render the explanation in the infoview.
+
+
+Example command usage:
+```
+#explain "This is an explanation."
+```
+This will render the provided text in the infoview.
+
+
+# Implementation
+
+This code uses `MarkdownDisplay` from `ProofWidgets`, and thus supports
+markdown and latex via mathjax.
+
+-/
+
+namespace Mathlib.Tactic
+
+open Lean Elab ProofWidgets ProofWidgets.Jsx
+
+/-- Displays the markdown source in `md` in a widget when the cursor is placed at `stx`. -/
+def displayMarkdown (md : String) (stx : Syntax) : CoreM Unit := do
+  let html : Html := <MarkdownDisplay contents={md}/>
+  Widget.savePanelWidgetInfo
+    (hash HtmlDisplayPanel.javascript)
+    (return json% { html: $(← Server.RpcEncodable.rpcEncode html) })
+    stx
+
+/-- Syntax for the `explain` tactic elaborator. -/
+syntax (name := explainTacStx) "explain" str ("in" ppIndent(tactic))? : tactic
```

Comment:
What do you think about it taking a tactic sequence instead of a single tactic? I think it's `tacticSeq` or `tacticSeqIndentGt`? (The second one might be better, since it's supposedly for nested by's.)

When you eval the tactic, it's possible that all you need to do is wrap it with `` `(by $seq:tacticSeq)`` and send it to `evalTactic`.